Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -13,6 +13,216 @@
+                }
+                h = static_cast<AZ::u16>(Result | Sign);
+            }
+
+            operator float() const
+            {
+                AZ::u32 Mantissa;
+                AZ::u32 Exponent;
+                AZ::u32 Result;
+
+                Mantissa = h & 0x03FF;
+
+                if ((h & 0x7C00) != 0)  // The value is normalized
+                {
+                    Exponent = ((h >> 10) & 0x1F);
+                }
+                else if (Mantissa != 0)     // The value is denormalized
+                {
+                    // Normalize the value in the resulting float
+                    Exponent = 1;
+
+                    do
+                    {
+                        Exponent--;
+                        Mantissa <<= 1;
+                    } while ((Mantissa & 0x0400) == 0);
+
+                    Mantissa &= 0x03FF;
+                }
+                else                        // The value is zero
+                {
+                    Exponent = static_cast<AZ::u32>(-112);
+                }
+
+                Result = ((h & 0x8000) << 16) | // Sign
+                    ((Exponent + 112) << 23) | // Exponent
+                    (Mantissa << 13);          // Mantissa
+
+                return *(float*)&Result;
+            }
+
+        private:
+            AZ::u16 h;
+        };
+
+        template <AZ::RHI::Format>
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::u32 RetrieveUintValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::s32 RetrieveIntValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <>
+        float RetrieveFloatValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0.0f;
+        }
+
+        template <>
+        AZ::u32 RetrieveUintValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0;
+        }
+
+        template <>
+        AZ::s32 RetrieveIntValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0;
+        }
+
+        float ScaleValue(float value, float origMin, float origMax, float scaledMin, float scaledMax)
+        {
+            return ((value - origMin) / (origMax - origMin)) * (scaledMax - scaledMin) + scaledMin;
+        }
+
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index, AZ::RHI::Format format)
+        {
+            switch (format)
+            {
+            case AZ::RHI::Format::R8_UNORM:
+            case AZ::RHI::Format::A8_UNORM:
```

Comment:
Something to think about with these switch statements, though maybe in a follow-up PR. Once you're supporting the component index, instead of taking in format directly here, you might want an intermediate format that's just "UNORM8", "SNORM8", "UNORM16", etc.